src/sp/it/pl/gui/objects/contextmenu/ContextMenuGenerator.kt

```diff
@@ -0,0 +1,85 @@
+package sp.it.pl.gui.objects.contextmenu
+
+import javafx.event.ActionEvent
+import javafx.scene.Node
+import javafx.scene.control.ContextMenu
+import javafx.scene.control.Menu
+import javafx.scene.control.MenuItem
+import sp.it.pl.gui.pane.collectionUnwrap
+import sp.it.pl.gui.pane.collectionWrap
+import sp.it.pl.util.collections.map.ClassListMap
+import sp.it.pl.util.dev.fail
+import sp.it.pl.util.functional.asArray
+import sp.it.pl.util.functional.asIf
+import sp.it.pl.util.functional.getElementType
+import sp.it.pl.util.functional.net
+import sp.it.pl.util.functional.seqOf
+import sp.it.pl.util.graphics.menuItem
+import sp.it.pl.util.graphics.menuSeparator
+
+val contextMenuGenerator = ContextMenuGenerator()
```

Comment:
couldn't ContextMenuGenerator be an object if you only use one global instance?

Reply:
It would be simpler, but that would prevent others using their own instances. There is a difference between providing a default implementation and forcing everyone use the same instance.